ivadomed/loader/dataset_group.py

```diff
@@ -0,0 +1,159 @@
+        self.train_filename_pairs: List[Tuple[list, list, str, dict]] = []
+        self.val_filename_pairs: List[Tuple[list, list, str, dict]] = []
+        self.test_filename_pairs: List[Tuple[list, list, str, dict]] = []
+
+        # DatasetGroup config
+        self.config = config
+
+        # This sets the expected number of files at the dataset group level. Useful for validating subsequent filesets and whether they are compliant or not.
+        self.n_expected_input: int = int(dict_file_group_spec.get(DataloaderKW.EXPECTED_INPUT, 0))
+        if self.n_expected_input <= 0:
+            raise ValueError("At DataSetGroup level, Number of Expected Input must be > 0")
+
+        self.n_expected_gt: int = int(dict_file_group_spec.get(DataloaderKW.EXPECTED_GT, 0))
+        if self.n_expected_gt <= 0:
+            raise ValueError("At DataSetGroup level, Number of Expected Ground Truth must be > 0")
+
+        # This is the key attribute that needs to be populated once data loading is complete.
+        if (ModelParamsKW.IS_2D not in config.model_params and not config.model_params.get(ModelParamsKW.IS_2D)):
+            self.validate_update_3d_train_val_test_and_filename_pairs(config, dict_file_group_spec)
+        else:
+            self.validate_update_2d_train_val_test_dataset_and_filename_pairs(config, dict_file_group_spec)
+
+        # Assign default labels if not provided.
+        self.name: str = dict_file_group_spec.get(DataloaderKW.DATASET_GROUP_LABEL, "DefaultDatasetGroupLabel")
+
+    def validate_update_3d_train_val_test_and_filename_pairs(self, config, dict_file_group_spec):
+        # Instantiate all the 3D Train/Val/Test Datasets
+        for a_train_dataset in dict_file_group_spec[DataloaderKW.TRAINING]:
+            if a_train_dataset[DataloaderKW.TYPE] == "FILES":
+                train_set = Files3DDataset(a_train_dataset, config)
+                self.train_dataset.append(train_set)
+                self.train_filename_pairs.extend(train_set.filename_pairs)
+        for a_val_dataset in dict_file_group_spec[DataloaderKW.VALIDATION]:
+            if a_val_dataset[DataloaderKW.TYPE] == "FILES":
+                val_set = Files3DDataset(a_val_dataset, config)
+                self.val_dataset.append(val_set)
+                self.val_filename_pairs.extend(val_set.filename_pairs)
+        for a_test_dataset in dict_file_group_spec[DataloaderKW.TEST]:
+            if a_test_dataset[DataloaderKW.TYPE] == "FILES":
+                test_set = Files3DDataset(a_test_dataset, config)
+                self.test_dataset.append(test_set)
+                self.test_filename_pairs.extend(test_set.filename_pairs)
+
+    def validate_update_2d_train_val_test_dataset_and_filename_pairs(self, config, dict_file_group_spec):
+        # Instantiate all the 2D Train/Val/Test Datasets
+        for a_train_dataset in dict_file_group_spec[DataloaderKW.TRAINING]:
+
+            if a_train_dataset[DataloaderKW.TYPE] == "FILES":
```

Comment:
Just for my understanding, in the long-term, we would have an `elif` for the REGEX path and `elif` for the BIDS path here?
Is that correct?

Reply:
Yes, precisely! That is the place where other path will be introduced.